daq/stream_monitor.py

```diff
@@ -149,16 +153,18 @@ def event_loop(self):
                 LOGGER.error('Monitoring exception in callback: %s', e)
                 LOGGER.exception(e)
             self.log_monitors()
-            self.dirty = False
-            fds = self.poller.poll(self.timeout_ms)
+            fds = self.poller.poll(self.timeout_sec * 10e3 if self.timeout_sec else None)
             LOGGER.debug('Monitoring found fds %s', fds)
             if fds:
                 for fd, event in fds:
-                    self.process_poll_result(event, fd)
-                    if self.dirty:
-                        LOGGER.debug('Monitoring set modified, re-issuing poll...')
-                        break
-            else:
-                LOGGER.debug('Monitoring no fds found')
-                return True
+                    if fd in self.callbacks: # Make sure it's still there.
+                        self.process_poll_result(event, fd)
+            # Check for timeouts
+            frozen_callbacks = copy.copy(self.callbacks)
+            for fd in frozen_callbacks:
+                name, _, _, on_error, timeout, _ = frozen_callbacks[fd]
+                if timeout and datetime.fromtimestamp(time.time()) >= timeout:
+                    LOGGER.error('Monitoring timeout fd %d (%s)', fd, name)
```

Comment:
add the actual timeout value to the logging message, I think that will be useful for diagnosing any problems